### PR TITLE
feat: edit community channels

### LINKED ISSUE #2344: Support editing & displaying channel descriptions

### PR DESCRIPTION
**Pull request by emizzle**
Closes #2344, #2658

Add ability to edit name, description, and private fields of a community channel.

feat: Display community channel description
Ensure the width of the description does not surpass the context menu and instead wraps to the next line.

feat: After channel is created, set it as the active channel

### NOTE
1. ~~The `private` field is currently a boolean to support the toggle that is currently in the UI. There is no functionality in status-go beyond storing the boolean value. The functionality in the design that supports private permissions will need to be added later.~~
2. ~~There is a [corresponding status-go PR for supporting the editing of channels](https://github.com/status-im/status-go/pull/2236). That PR should be merged first, then this PR will need to be updated with the new commit BEFORE this is merged.~~

### CODE REVIEW

File: src/app/chat/view.nim [hunk 1]
```diff
@@ -996,3 +998,23 @@ QtObject:
   proc formatInputCode(self: ChatsView, inputText: string): string {.slot.} =
     let strikeThroughRegex = re"""(?<!\>)`(?!<span style=" font-family:'monospace';">)([^*]+)(?!<\/span>)`"""
     self.formatInputStuff(strikeThroughRegex, inputText)
+
+  proc createCommunityChannel*(self: ChatsView, communityId: string, name: string, description: string, private: bool): string {.slot.} =
```
Comment (jrainville) [inline src/app/chat/view.nim:1002]: Why was this moved from `views/communities` to  `view.nim`?

**Review by jrainville — COMMENTED** [review]
Looks good.
For the private field, I'm not even sure if it will ever be used. It was part of the design, but the specs keep changing and with phase 3 using tokenization, no idea if it will be still. Anyway, we can keep it for now, but it might be removed.

**Comment by guylouis** [comment]
I agree that the private field should be removed for now
@jrainville 
@emizzle

**Comment by anastasiyaig** [comment]
@emizzle could you please limit the rights to edit channels to admin only?

File: src/app/chat/view.nim (hunk 1 above)
Comment (emizzle) [reply src/app/chat/view.nim:1002]: This was moved so that we could call `setActiveChannel` on the created/edited channel in the chats view (`view.nim`). The only way to achieve this from within the communities view (`views/communities.nim`) was to use the event emitter to emit an event from the communities view, but that can get messy and I would prefer to avoid use of the event emitter if possible.

**Comment by emizzle** [comment]
@guylouis - the private channel has been removed.

@anastasiyaig - the edit channel rights have been limited to admin only (the Edit Channel action will only appear in the context menu if the user is a community admin).

**Comment by emizzle** [comment]
@iurimatias waiting to merge because the status-go branch hasn't been merged yet.

**Comment by vkjr** [comment]
@emizzle, feel free to merge status-go branch,it is ready and approved

**Comment by anastasiyaig** [comment]
> @guylouis - the private channel has been removed.

> @anastasiyaig - the edit channel rights have been limited to admin only (the Edit Channel action will only appear in the context menu if the user is a community admin).

@emizzle great work, lets merge it!

**Review by jrainville — APPROVED** [review]

**Comment by emizzle** [comment]
We need to wait until https://github.com/status-im/status-go/pull/2236 is merged.

**Comment by vkjr** [comment]
> We need to wait until https://github.com/status-im/status-go/pull/2236 is merged.

@emizzle, feel free to merge it (with version bumped) if it was tested on desktop. (mobile pr created today and not tested yet.

**Comment by emizzle** [comment]
@vkjr thank you so much. I would have merged it, but there are two things blocking the merge: 1 CodeClimate issue and a failed build.

**Comment by vkjr** [comment]
@emizzle, codeclimate issue related to autogenerated file so can be ignored. Build issue wasn't a problem with the build itself, but with the upload step (something wrong with Digital Ocean). Now it works, so merged it.

**Comment by richard-ramos** [comment]
"Edit channel" is being added for normal chats after activating a community and returning to chat module

https://user-images.githubusercontent.com/1106587/120528998-06bf7780-c3aa-11eb-8d12-65de6668f012.mp4

**Comment by iurimatias** [comment]
@emizzle please add the new api calls to the docs as well https://status-im.github.io/status-desktop/api/statusgo/communities/ just the method name and parameters will do

**Comment by iurimatias** [comment]
the user docs and dev guide as well

**Comment by anastasiyaig** [comment]
@emizzle would u mind to rebase this one? I want to test it with mobile alltogether, but it has a crash on communities which is fixed in master already. Also, please don't merge it

**Comment by emizzle** [comment]
> "Edit channel" is being added for normal chats after activating a community and returning to chat module

@richard-ramos fixed.

> @emizzle please add the new api calls to the docs as well https://status-im.github.io/status-desktop/api/statusgo/communities/ just the method name and parameters will do
> the user docs and dev guide as well

@iurimatias done.

> @emizzle would u mind to rebase this one?

@anastasiyaig done.

**Comment by anastasiyaig** [comment]
@emizzle this branch does contain the crash for communities, which was already fixed for master https://github.com/status-im/status-desktop/issues/2628 , i can't reproduce the crash in recent master, but it constantly crashes in current branch. Not sure what is the root cause, just dont want to bring this crash back again to master. Please take a look: it crashes when creating community, adding members, or channel

**Comment by emizzle** [comment]
That fix is definitely in this branch as it was rebased on top of master. But i've added additional null-checking code to prevent a crash. I haven't really tested it thoroughly because I don't know exact reproduction steps.

**Comment by anastasiyaig** [comment]
please merge this PR

NOTE: some regression appeared, we lost channel desc in community but it is already in master https://github.com/status-im/status-desktop/issues/2658

**Comment by emizzle** [comment]
Community channel description is not showing for this PR?

**Comment by anastasiyaig** [comment]
@emizzle is is shown when creating a channel for community and editing however viewing a channel does not show it

Screenshot 2021-06-08 at 10 37 35

**Comment by anastasiyaig** [comment]
thanks @emizzle for fixing https://github.com/status-im/status-desktop/issues/2658 as well here in this branch
